src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3743,21 +3794,24 @@ NetworkOPsImp::addAccountHistoryJob(SubAccountHistoryInfoWeak subInfo)
                             send(rpcError(rpcINTERNAL), true);
                             return;
                         }
-                        Json::Value jvTx = transJson(
-                            *stTxn, meta->getResultTER(), true, curTxLedger);
-                        jvTx[jss::meta] = meta->getJson(JsonOptions::none);
-                        jvTx[jss::account_history_tx_index] = txHistoryIndex--;
 
+                        auto const mRef = std::ref(*meta);
+                        auto const trR = meta->getResultTER();
+                        MultiApiJson jvTx(
+                            {transJson(stTxn, trR, true, curTxLedger, mRef, 1),
+                             transJson(
+                                 stTxn, trR, true, curTxLedger, mRef, 2)});
+
+                        jvTx.set(
+                            jss::account_history_tx_index, txHistoryIndex--);
                         if (i + 1 == num_txns ||
                             txns[i + 1].first->getLedger() != tx->getLedger())
-                            jvTx[jss::account_history_boundary] = true;
+                            jvTx.set(jss::account_history_boundary, true);
 
-                        RPC::insertDeliveredAmount(
-                            jvTx[jss::meta], *curTxLedger, stTxn, *meta);
                         if (isFirstTx(tx, meta))
                         {
-                            jvTx[jss::account_history_tx_first] = true;
-                            send(jvTx, false);
+                            jvTx.set(jss::account_history_tx_first, true);
+                            sendMultiApiJson(jvTx, false);
```

Comment:
We did not do logging before. Logging on networking operations is not a great idea, since it can potentially flood the logs e.g. on a persistent network failure. It's better to have metrics of this kind, but we did not have one here either. If we want it, it would belong to a different PR.